packages/renovate/project.bri

```diff
@@ -1,19 +1,74 @@
 import * as std from "std";
-import { npmInstallGlobal } from "nodejs";
+import nodejsRecipe from "nodejs";
+import pnpmRecipe, { pnpmInstall } from "pnpm";
+import python from "python";
+import { runNushell } from "nushell";
 
 export const project = {
   name: "renovate",
   version: "41.1.4",
-  extra: {
-    packageName: "renovate",
-  },
+  repository: "https://github.com/renovatebot/renovate.git",
 };
 
+const source = Brioche.gitCheckout({
+  repository: project.repository,
+  ref: project.version,
+});
+
+// Pin to Node.js v22
+const nodejs = nodejsRecipe({ version: "22" });
+const pnpm = pnpmRecipe({ nodejs });
+
 export default function renovate(): std.Recipe<std.Directory> {
-  return npmInstallGlobal({
-    packageName: project.extra.packageName,
-    version: project.version,
-  }).pipe((recipe) => std.withRunnableLink(recipe, "bin/renovate"));
+  // Install dependencies using pnpm
+  // TODO: Can we get rid of dev dependencies (at least for the final output)?
+  const npmPackage = pnpmInstall({
+    source,
+    dependencies: [python],
+    nodejs,
+  });
+
+  // Run the Renovate build. We explicitly update the config to use the
+  // provided version of pnpm instead of trying to download Renovate's
+  // pinned version.
+  const dist = std.runBash`
+    pnpm --config.manage-package-manager-versions=false config set manage-package-manager-versions=false
```

Comment:
By default, `pnpm` tries to update itself to a new version specified by the workspace, but this fails (unless networking is enabled for the build). Annoyingly, the `pnpm config` command _also_ tries to update pnpm (which is why we also need the `--config` flag).

I felt better about using our managed version of pnpm rather than downloading the version specified by Renovate, but I don't feel strongly about this. I'd be fine in the future with just enabling networking for this build via `.unsafe(...)`

Reply:
I prefer as you did, this way we fully control the pnpm version used. Preferable to be something Brioche managed than relying on an external binary.